### ISSUE TITLE
How to access pbcopy/pbpaste if I run the tmux with other user?

### ISSUE THREAD

**Issue by guanghechen** [ChrisJohnsen/tmux-MacOSX-pasteboard#80]
Hi @ChrisJohnsen , thank you for make such a great tool! I have encountered an edge situation that cannot access the system clipboard correctly and hope to get a suggestion to resolve it. 

Repro steps

1. Login as user 'alice'.
2. Open a terminal and change the user : 'sudo su bob'.
3. Run tmux as 'bob': 'tmux'.

Then I lose the access to the system clipboard.

**Comment by ChrisJohnsen** [comment]
Sorry, I don't have a Mac system to test with anymore.

My impression is that—without ever involving *tmux*—different users would have different pasteboards.

If `pbpaste` run from the Bob shell (immediately after step 2—without *tmux* involved) does not show the contents of Alice's pasteboard, then each user probably has separate pasteboards.

---

I suspect that the relevant structure looks like this:
* global bootstrap namespace
    *  Alice's per-user bootstrap namespace
        * Alice's GUI login session bootstrap namespace
            * Alice's pasteboard server (used to be the `pbs` process, not sure if it still is)
            * the first shell started in Alice's Terminal window
    * Bob's per-user bootstrap namespace
        * the Bob shell created by `sudo su bob`

Note that the Bob shell ends up in Bob's bootstrap namespace hierarchy despite having been created from "inside" Alice's GUI session.

An old document that describes this is [TN2083][tn2083]. When last I tried, you could sort of verify this structure by examining various kinds of `launchctl` output. But this stuff seems to only be lightly documented and has changed somewhat over various major MacOS releases.

[tn2083]: https://developer.apple.com/library/archive/technotes/tn2083/_index.html#//apple_ref/doc/uid/DTS10003794-CH1-SUBSECTION10

Looking up various services (like the pasteboard) searches *up* the hierarchy of namespaces, but never across to other namespaces. 

Alice's shell (and its children) can only find services in
* Alice's GUI login session bootstrap namespace (where the pasteboard is registered),
* Alice's per-user bootstrap namespace,
* and the global bootstrap namespace.

The Bob shell (and its children) can only find services in
* Bob's per-user bootstrap namespace,
* and the global bootstrap namespace.

---

The wrapper program from this repository does switch to different namespaces, but only in the case of moving from a user's per-user bootstrap namespace to the *same* user's GUI login bootstrap namespace. Letting a user access another user's bootstrap namespaces would probably be a big security problem (lots more than just the pasteboard goes through this bootstrap namespace system). I don't think there is an exception for the "created from inside another user's GUI session" scenario. If there is such an exception (i.e. Bob's shell can access Alice's pasteboard before running *tmux*), I don't know of a way to leverage it to maintain that access inside a *tmux* session.

**Comment by guanghechen** [comment]
Hi ChrisJohnsen,

Thank you very much for your patient reply and detailed explanation. I actually lack knowledge in this area (macOS system APIs and the C language).

Based on my practical experience, after switching to bob in the second step, I can still access alice's clipboard, and even when I enter nvim at this point, I can still access alice's system clipboard, so I naturally assumed they were shared. On the other hand, when alice opens the terminal and directly runs tmux and then switches to bob in one of the tmux panes, they can still access the system clipboard. However, my requirement, as mentioned earlier in this issue, is to start tmux as bob (so that each new pane opened is still under Bob's identity without needing to execute the su command) and be able to normally access the system clipboard.

In summary, I believe that different users on macOS can share the system clipboard. I'm not sure if this information is useful to you, but in any case, thank you very much for your help. 🙏

**Comment by guanghechen** [comment]
Finally, I used a hack method to achieve this：

1. Alice start a process to watch a file change and auto read the content and write in to the pbcopy.

2. Bob use a fake clipboard which read/write content from the special file. Also bob can config a fake clipboard provide to access the special file.

3. For paste, using the alacritty paste action works fine for this case.
